frontend/src/pages/keysign/JoinKeysignFlow.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useRef, useState } from 'react';
+import JoinKeysign from '../../components/keysign/JoinKeysign';
+import NavBar from '../../components/navbar/NavBar';
+import { useTranslation } from 'react-i18next';
+import { joinSession } from '../../services/Keygen/Keygen';
+import { useNavigate, useParams } from 'react-router-dom';
+import { storage } from '../../../wailsjs/go/models';
+import { useQueryClient } from '@tanstack/react-query';
+import { GetVault } from '../../../wailsjs/go/storage/Store';
+import {
+  KeysignMessage,
+  KeysignPayload,
+} from '../../gen/vultisig/keysign/v1/keysign_message_pb';
+import DiscoveryServiceScreen from '../../components/keygen/DiscoveryService';
+import KeysignError from '../../components/keysign/KeysignError';
+import KeysignView from '../../components/keysign/KeysignView';
+import { KeysignPayloadUtils } from '../../extensions/KeysignPayload';
+import { useWalletCore } from '../../main';
+import { Endpoint } from '../../services/Endpoint';
+import KeysignDone from '../../components/keysign/KeysignDone';
+const JoinKeysignFlow: React.FC = () => {
+  const { t } = useTranslation();
+  const queryClient = useQueryClient();
+  const [currentScreen, setCurrentScreen] = useState<number>(0);
+  const [localPartyID, setLocalPartyID] = useState<string>('');
+  const serverURL = useRef<string>('');
+  const serviceName = useRef<string>('');
+  const sessionID = useRef<string>('');
+  const hexEncryptionKey = useRef<string>('');
+  const currentVault = useRef<storage.Vault>(undefined as any);
+  const { publicKeyECDSA } = useParams<{ publicKeyECDSA: string }>();
+  const [keysignErr, setKeysignErr] = useState<string>('');
+  const walletCore = useWalletCore();
+  const [messagesToSign, setMessagesToSign] = useState<string[]>([]);
+  const [keysignPayload, setKeysignPayload] = useState<KeysignPayload>();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    setCurrentScreen(0);
+  }, []);
+  useEffect(() => {
+    const processJoinKeysign = async () => {
+      if (publicKeyECDSA === undefined) {
+        throw new Error('publicKeyECDSA is undefined');
+      }
+      const keysignMessage = queryClient.getQueryData<KeysignMessage>([
+        'keysignMessage',
```

Comment:
If there could be different keysign message, then the query key should be unique to not accidently use the old message in a new keysign operation. E.g. query key would be `['keysignMessage, someKeysignMessageId]`